### ISSUE TITLE
Barrierfree calender

### ISSUE THREAD

**Issue by MarauderXtreme** [fsr/ese-webseite#15]
We should create the barrierfree version of the calender. We have our awesome events.yaml so we just need a proper layout.

**Comment by JustusAdam** [comment]
You can either do it yourself, or make me a draft of how it should look, and I'll add the logic.

**Comment by MarauderXtreme** [comment]
I think the logic should be sufficient by now. I added it this way because I want to hide the timetable on small devices because it overflows. Because of that it is a bit hard to correctly markup these things to get them to work.
Maybe you have a better idea.

**Comment by sjmielke** [comment]
1. You hide the timetable on mobile devices? ...what? I mean, okay, making the barrier-free version the first option is probably a good idea, the big table isn't so great for mobile devices, but there is no way to access it at all? The way you start spamming JS magic you can surely find a solution that makes it possible for mobile users to simply use the reversed order.
2. I know barrier-free and noscript-compatible are two very different things but it feels very very wrong to me that I have to use JS to get to a barrier-free version.
3. I don't know how screenreaders work in detail and how well they can deal with these accordions, but if we really look at the plain HTML, putting the non-barrier-free one first and not offering a skip link (or at least I could not find one, but skimming 15000 lines [2015] is a little bit more difficult than skimming 300 lines [2014]) sounds like a bad idea.

So, my request: change places and make only the desktop version show the then second variant (full table) first (but not by using JS and thereby making the full table completely unaccessible without JS).

**Comment by MarauderXtreme** [comment]
I never was a fan of this table. In my opinion the list is enough. But they wanted it, so we did it.    
Because of the actual layout of the website I have seen no other possibility of bringing it to the site. If you know how to bring those things properly to the website feel free to make a PR. 

The accordion is probably better [accesible](http://foundation.zurb.com/docs/components/accordion.html#accessibility) through a screen reader than other things.

The skip link was introduced with #9 but it skips to the website content.

The line problem is described [here](https://github.com/fsr/ese-webseite/issues/17#issuecomment-141775119).

**Comment by sjmielke** [comment]
> I never was a fan of this table.

I'd be inclined to say that presenting a *timetable* like a *table* is a concept that has aged very well and resonates with a lot of people. I see no reason to exclude it from mobile. Just use some JS to expand the second one (containing the list) as a standard or something.

> The skip link was introduced with #9 but it skips to the website content.

I actually found that one, but if there is ARIA markup for the accordion, that solves my problem, too, I guess.

> I have seen no other possibility of bringing it to the site

With the screenreader stuff out of the way (maybe a link before the table would be nice to inform people that there is a list variant cleverly hidden below the other plan), my only complaint is that the accordion stays with the first element open and the second closed, if no JS is enabled. But I agree that this is probably a bit tricky to circumvent and any "solution" will be an ugly hack.

**Comment by MarauderXtreme** [comment]
The problem with tables in HTML is that they are not very responsive (uncool for a mobile layout) and are a pain in the arse with making the accessible. I tried it once and did not go well. Bacause of that I dislike any kind of table on website that are not easy as pie. Certainly a timetable with free spaces is not.
Bringing it to a correct table markup was necessary because fixing heights and widths to prevent weird overflowing and floating would not have worked out.
Bringing a picture instead would be easier but one does not simply want this.

You can add the barierfree version as an own site but than you can not have the timetable on /  on mobile devices. Rather you need to put a link there. This would fix problems but you would have drawbacks.

**Comment by sjmielke** [comment]
Are we still going to get a external-HTML-file-link for mobile users or will "try guessing that there is one if you request the desktop version" stay the only option?

Also there are a few other things that I dodn't want to create a new Misc-Issue for:
* the width of the text could be a bit more on mobile - I think someone already mentioned this, was there a decision or a reason for this current behavior? Just a little more would be nice, I think.
* h3s don't stand out enough, that makes reading the other pages a bit difficult. They looked different on @heisigs machine, but they look very non-heading-y on both my laptop and my smartphone. (as already mentioned in the "locked" issue #21).
* Can we get rid of the Special-News-Box if there are no news? First, because its utterly useless, second, because if the viewer now gets used to the box, he won't realize when something else will later be in there - if it wasn't visible now, it would really be registered as some important new update when we start using it.

By the way, all in all the site turned out pretty well by now, I think, the hard work of the last weeks has definitely paid off :)

**Comment by MarauderXtreme** [comment]
> the width of the text could be a bit more on mobile - I think someone already mentioned this, was there a decision or a reason for this current behavior? Just a little more would be nice, I think.

I will adress this with #21. This is related to the usage of the basic jekyll styles that somehow behave not like "vanilla" foundation would.
> h3s don't stand out enough

I think this has the same reason as the width of the text. I will adress this too in #21.
> Can we get rid of the Special-News-Box if there are no news?

Agree. This is a good idea. Maybe @JustusAdam could do this because I don't know if would do this correct with the liquid templating.

> Are we still going to get a external-HTML-file-link
Definitely not. As already stated many times no one with a small/medium smartphone wants and expects a huge table he has to scroll through both vertically and horizontally. It is not a good optimization because you can not see the structure with the first look on it. It is a very special layout with rowspans that prevent an easy information gain from it on a small device.
So I will not provide a link that will distract the visitor from the correct/easier layout.
As you said you are always able to request the desktop version on you smartphone which will display the whole table.

**Comment by JustusAdam** [comment]
@MarauderXtreme @sjmielke This doe snot belong in this issue. I'll delete both your comments, move their contents to another issue please.

**Comment by MarauderXtreme** [comment]
My opinion. But @sjmielke wrote here about this so I wanted to answer his suggestion since he seems not to read every issue.

**Comment by MarauderXtreme** [comment]
Oh wait. We maybe should discuss the proper aria labelling over here.

**Comment by JustusAdam** [comment]
What do you mean?

**Comment by MarauderXtreme** [comment]
We probably need to add some aria labels or other attributes to the table because of the structure. We should provide the best accessible markup for all of our visitors.

**Comment by sjmielke** [comment]
I just noticed that the barrierfree timetablelist colors all entries green - not only in german but also (all entries!) in english. Something wrong there?

**Comment by JustusAdam** [comment]
No? On my machine al is normal.

**Comment by sjmielke** [comment]
Oops, forgot to mention I meant the mobile layout. (edited it accordingly)

**Comment by JustusAdam** [comment]
I see it ... where does that come from?

**Comment by JustusAdam** [comment]
I found the source, I am waiting on some information on why it was implemented.

**Comment by MarauderXtreme** [comment]
@JustusAdam implemented some information about events with English "support". While refactoring I tried to reimplement this function but I think I forgot to properly write this. With the previous commit, it should be done.

**Comment by sjmielke** [comment]
Kleines Zitat aus dem, was bis jetzt in der ESE-EVA reingekommen ist:

> Der Zeitplan zur ESE auf der Website wurde auf mobilen Geräten nicht angezeigt - Responsive Design ist ja cool, aber bitte testet auch was ihr macht ;-)

:)

**Comment by MarauderXtreme** [comment]
Well. How many times this came in a EVA?

**Comment by MarauderXtreme** [comment]
See ***THE TABLE*** #33
